plugins/org.jkiss.dbeaver.ext.cubrid/src/org/jkiss/dbeaver/ext/cubrid/model/CubridObjectContainer.java

```diff
@@ -0,0 +1,324 @@
+		final GenericMetaObject tableObject;
+		final GenericMetaObject columnObject;
+
+		protected CubridTableCache(CubridDataSource dataSource) {
+			super(GenericUtils.getColumn(dataSource, CubridConstants.OBJECT_TABLE, JDBCConstants.TABLE_NAME));
+			this.dataSource = dataSource;
+			this.tableObject = dataSource.getMetaObject(CubridConstants.OBJECT_TABLE);
+			this.columnObject = dataSource.getMetaObject(CubridConstants.OBJECT_TABLE_COLUMN);
+			setListOrderComparator(DBUtils.<CubridTableBase>nameComparatorIgnoreCase());
+		}
+
+		public CubridDataSource getDataSource() {
+			return dataSource;
+		}
+
+		@Override
+		public @NotNull JDBCStatement prepareLookupStatement(@NotNull JDBCSession session, @NotNull CubridObjectContainer owner,
+			@Nullable CubridTableBase object, @Nullable String objectName) throws SQLException {
+			return dataSource.getMetaModel().prepareTableLoadStatement(session);
+		}
+
+		@Override
+		protected CubridTableColumn fetchChild(@NotNull JDBCSession arg0, @NotNull CubridObjectContainer owner,
+			@NotNull CubridTableBase table, @NotNull JDBCResultSet dbResult) throws SQLException, DBException {
+			return new CubridTableColumn(table, dbResult);
+		}
+
+		@Override
+		protected JDBCStatement prepareChildrenStatement(@NotNull JDBCSession session, @NotNull CubridObjectContainer owner,
+			@Nullable CubridTableBase forTable) throws SQLException {
+			return dataSource.getMetaModel().prepareTableColumnLoadStatement(session, owner, forTable);
+		}
+
+		@Override
+		protected @Nullable CubridTableBase fetchObject(@NotNull JDBCSession session, @NotNull CubridObjectContainer owner,
+			@NotNull JDBCResultSet dbResult) throws SQLException, DBException {
+			return getDataSource().getMetaModel().createTableImpl(session, owner, tableObject, dbResult);
+		}
+
+	}
+
+	public class CubridSystemTableCache extends CubridTableCache {
+
+		protected CubridSystemTableCache(CubridDataSource dataSource) {
+			super(dataSource);
+		}
+
+		@Override
+		public @NotNull JDBCStatement prepareLookupStatement(@NotNull JDBCSession session, @NotNull CubridObjectContainer owner,
+			@Nullable CubridTableBase object, @Nullable String objectName) throws SQLException {
+			return dataSource.getMetaModel().prepareSystemTableLoadStatement(session, owner, object, objectName);
+		}
+
+	}
+
+	public class CubridSystemViewCache extends CubridTableCache {
+
+		protected CubridSystemViewCache(CubridDataSource dataSource) {
+			super(dataSource);
+		}
+
+		@Override
+		public @NotNull JDBCStatement prepareLookupStatement(@NotNull JDBCSession session, @NotNull CubridObjectContainer owner,
+			@Nullable CubridTableBase object, @Nullable String objectName) throws SQLException {
+			return dataSource.getMetaModel().prepareSystemViewLoadStatement(session, owner, object, objectName);
+		}
+
+	}
+
+	public class CubridIndexCache extends JDBCCompositeCache<CubridObjectContainer, CubridTableBase, CubridTableIndex, GenericTableIndexColumn>{

```

Comment:
I do not see any specific reason for CubridTableIndex to exist. Why can't you just use the Generic index?

Reply:
I want to override **prepareObjectsStatement** to support separate index based on owner. but generic's indexcache is not allow to override it from meta-model class.

Reply:
Ok.